internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -277,6 +274,9 @@ func (c *Coordinator) Upgrade(ctx context.Context, version string, sourceURI str
 		}
 	}
 
+	if c.state.overrideState != nil && c.state.overrideState.state == agentclient.Upgrading {
+		return fmt.Errorf("%s already in progress", c.state.overrideState.message)
```

Comment:
if you initiate 2 upgrades in a row from fleet
first one is kicked off second one is returning this error. will this result in Unhealthy state in fleet UI?

Reply:
I've changed this, so if you don't get an error if you initiate the second upgrade via fleet.  you do get the error message if you initiate via CLI.  All upgrades get ACKed together if there is an error, if successful the reboot ACKs.